app/assets/javascripts/miq_application.js

```diff
@@ -1405,7 +1405,12 @@ function miqToolbarOnClick(e) {
     if (typeof params == "undefined") {
       miqJqueryRequest(tb_url, {beforeSend: true, complete: true});
     } else {
-      miqJqueryRequest(tb_url, {beforeSend: true, complete: true, data: params});
+      options = {beforeSend: true, data: params};
```

Comment:
Actually, @bmclaughlin , the only difference between the 2 outer branches here is complete:true.. maybe we could refactor 

```js
  if ((button.attr('name') == "perf_reload") ||
      (button.attr('name') == "vm_perf_reload") ||
      (button.attr('name').match("_console$"))) {
    if (typeof params == "undefined") {
      miqJqueryRequest(tb_url, {beforeSend: true});
    } else {
      miqJqueryRequest(tb_url, {beforeSend: true, data: params});
    }
  } else {
    if (typeof params == "undefined") {
      miqJqueryRequest(tb_url, {beforeSend: true, complete: true});
    } else {
      // +your changes here
      miqJqueryRequest(tb_url, {beforeSend: true, complete: true, data: params});
    }
  }
  return false;
```

to something like

```js
var no_complete = _.includes([
  'perf_reload',
  'vm_perf_reload',
  'download_choice__render_report_csv',
  'download_choice__render_report_pdf',
  'download_choice__render_report_txt',
], button.attr('name')) || button.attr('name').match(/_console$/);

var options = {
  beforeSend: true,
  complete: ! no_complete,
  data: params,
};

miqJqueryRequest(tb_url, options);
return false;
```

what do you think?